docs/csharp/fundamentals/exceptions/creating-and-throwing-exceptions.md

```diff
@@ -44,15 +44,47 @@ The following list identifies practices to avoid when throwing exceptions:
 - Don't throw <xref:System.Exception?displayProperty=nameWithType>, <xref:System.SystemException?displayProperty=nameWithType>, <xref:System.NullReferenceException?displayProperty=nameWithType>, or <xref:System.IndexOutOfRangeException?displayProperty=nameWithType> intentionally from your own source code.
 - Don't create exceptions that can be thrown in debug mode but not release mode. To identify run-time errors during the development phase, use Debug Assert instead.
 
-## Defining Exception Classes
+## Throw exceptions from task-returning methods
+
+Methods that return <xref:System.Threading.Tasks.Task%601> objects have some special considerations when it comes to exceptions. Exceptions that are thrown outside of the async state machine, that is, before `await` is called, emerge synchronously from the method, just like a non task-returning method. However, exceptions that are thrown within the async state machine are stored in the task and don't emerge until, for example, the task is awaited. For more information about stored exceptions, see [Asynchronous exceptions](../../asynchronous-programming/index.md#asynchronous-exceptions).
```

Comment:
This isn't quite right. See [Sharplabl](https://sharplab.io/#v2:EYLgtghglgdgNAFxAJwK7wCYgNQB8ACATAAwCwAUEQIwX7EAE+VArANwW1UCcAFAESAeDcAg+3wCU7cgDcIyegnoBeegGFkAUwgI1AZQSa1AWQgBjABaw1PcZ159VGrfQDOex5DMX6xiABsfajDEJGx4EADpdTVQna3JaLjlgym5+fAS9JwBrLwB7MAAHfwQoHJggjkoADkYANhV1fUitIw8YS1EKAG8Kel65U2QcgHd6NpGAUQAPYzV84tL+BAHh+mA1ADMc9UYucvI+ndqwgE0oNR8MKwkAXwqDoA=)

Here's how I'd word it:

```suggestion
Methods declared with the `async` modifier have some special considerations when it comes to exceptions. Exceptions thrown in an async method are stored in the returned task and don't emerge until, for example, the task is awaited. For more information about stored exceptions, see [Asynchronous exceptions](../../asynchronous-programming/index.md#asynchronous-exceptions).
```